src/Groups/directproducts.jl

```diff
@@ -44,19 +44,22 @@ julia> elements(G)
 ```
 """
 function direct_product(L::AbstractVector{<:GAPGroup}; morphisms=false)
-   X = GAP.Globals.DirectProduct(GapObj([G.X for G in L]))
-   DP = DirectProductGroup(X,L,X,true)
-   if morphisms
-      emb = [GAPGroupHomomorphism(L[i],DP,GAP.Globals.Embedding(X,i)) for i in 1:length(L)]
-      proj = [GAPGroupHomomorphism(DP,L[i],GAP.Globals.Projection(X,i)) for i in 1:length(L)]
-      return DP, emb, proj
-   else
-      return DP
-   end
+  @req length(L) > 0 "the collection of groups must be non-empty"
+  X = GAP.Globals.DirectProduct(GapObj([G.X for G in L]))
+  DP = DirectProductGroup(X, L, X, true)
+  if morphisms
+    emb = [GAPGroupHomomorphism(L[i], DP, GAP.Globals.Embedding(X, i)) for i in 1:length(L)]
+    proj = [
+      GAPGroupHomomorphism(DP, L[i], GAP.Globals.Projection(X, i)) for i in 1:length(L)
+    ]
+    return DP, emb, proj
+  else
+    return DP
+  end
 end
 
-function direct_product(L::GAPGroup...; morphisms=false)
-   return direct_product([x for x in L]; morphisms=morphisms)
+function direct_product(L::GAPGroup, Ls::GAPGroup...; morphisms=false)
+  return direct_product([L, Ls...]; morphisms=morphisms)
```

Comment:
Just for curiosity:
Why is the additional `GAPGroup` argument needed?
When the `direct_product(L::GAPGroup...)` method gets called then `L` is automatically nonempty.
(I could think of the idea that one wants to admit `direct_product()` via a dedicated method.)

Reply:
When you currently try to call `direct_product()` you get an ambiguity error, since all of the dispatches of the form `direct_product(::SomeType...)` are equally good.

This can only be resolved by changing all of these to take at least one argument of their preferred type.